src/material.js

```diff
@@ -174,6 +191,19 @@ export class Material {
         return new Lattice(this.lattice);
     }
 
+    getInchiStringForHash() {
+        const derivedProperties = this.getDerivedProperties();
+        if (derivedProperties.length > 0) {
+            const inchi = derivedProperties ? this.getDerivedPropertyByName('inchi') : null;
```

Comment:
There's no need for this check. The default value is an empty array which is a truth value